system-test/datastore.ts

```diff
@@ -704,6 +710,13 @@ describe('Datastore', () => {
       assert.strictEqual(secondEntities!.length, 3);
     });
 
+    it('should query the datastore with snapshot read', async () => {
```

Comment:
is this related to SUM/AVG changes? If not, let's move it into a separate PR

Reply:
ok